src/proto/h1/role.rs

```diff
@@ -328,16 +392,42 @@ impl Http1Transaction for Server {
         let init_cap = 30 + msg.head.headers.len() * AVERAGE_HEADER_SIZE;
         dst.reserve(init_cap);
         if msg.head.version == Version::HTTP_11 && msg.head.subject == StatusCode::OK {
+            if cfg!(feature = "layers") {
+                tracing::Span::current().record("http.flavor", &1.1);
+            }
             extend(dst, b"HTTP/1.1 200 OK\r\n");
         } else {
             match msg.head.version {
-                Version::HTTP_10 => extend(dst, b"HTTP/1.0 "),
-                Version::HTTP_11 => extend(dst, b"HTTP/1.1 "),
+                Version::HTTP_10 => {
+                    if cfg!(feature = "layers") {
+                        tracing::Span::current().record("http.flavor", &1.0);
+                    }
+                    extend(dst, b"HTTP/1.0 ");
+                }
+                Version::HTTP_11 => {
+                    if cfg!(feature = "layers") {
+                        tracing::Span::current().record("http.flavor", &1.1);
+                    }
+                    extend(dst, b"HTTP/1.1 ");
+                }
                 Version::HTTP_2 => {
-                    debug!("response with HTTP2 version coerced to HTTP/1.1");
+                    if cfg!(feature = "layers") {
+                        tracing::Span::current().record("http.flavor", &1.1);
+                        otel_status_messages
+                            .push("response with HTTP2 version coerced to HTTP/1.1".to_string());
+                    }
                     extend(dst, b"HTTP/1.1 ");
                 }
-                other => panic!("unexpected response version: {:?}", other),
+                other => {
+                    if cfg!(feature = "layers") {
+                        // 0.0 is a placeholder until `impl Debug for Version`
+                        tracing::Span::current().record("http.flavor", &0.0);
+                        tracing::Span::current().record("otel.status", &format!("{:?}", opentelemetry::trace::StatusCode::Error).as_str());
```

Comment:
again, we probably want to avoid getting the current span twice if we don't have to...